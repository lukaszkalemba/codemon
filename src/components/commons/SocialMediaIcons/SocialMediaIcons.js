import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'components/commons/Icon';
import facebook from 'assets/svgs/icon_facebook.svg';
import instagram from 'assets/svgs/icon_instagram.svg';
import dribbble from 'assets/svgs/icon_dribbble.svg';
import linkedin from 'assets/svgs/icon_linkedin.svg';
import S from './SocialMediaIcons.styles';

const SocialMediaIcons = ({ className }) => (
  <S.Wrapper className={className}>
    <S.A href="https://facebook.com" target="__blank">
      <Icon src={facebook} />
    </S.A>
    <S.A href="https://instagram.com" target="__blank">
      <Icon src={instagram} />
    </S.A>
    <S.A href="https://dribbble.com" target="__blank">
      <Icon src={dribbble} />
    </S.A>
    <S.A href="https://linkedin.com" target="__blank">
      <Icon src={linkedin} />
    </S.A>
  </S.Wrapper>
);

SocialMediaIcons.propTypes = {
  className: PropTypes.string.isRequired,
};

export default SocialMediaIcons;
