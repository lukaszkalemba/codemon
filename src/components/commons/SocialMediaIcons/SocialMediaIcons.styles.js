import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
`;

const A = styled.a`
  opacity: 0.5;
  transition: opacity 400ms ease-in-out;
  padding: 8px 0;

  :hover {
    opacity: 1;
  }
`;

export default { Wrapper, A };
