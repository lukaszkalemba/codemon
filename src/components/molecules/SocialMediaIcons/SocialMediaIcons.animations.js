const getWrapper = animate => {
  return {
    variants: {
      animate: {
        opacity: 1,
        y: 0,
        transition: { duration: 0.8, delay: 0 },
      },
      initial: { opacity: 0, y: 20 },
    },
    animate,
    initial: 'initial',
  };
};

export default { getWrapper };
