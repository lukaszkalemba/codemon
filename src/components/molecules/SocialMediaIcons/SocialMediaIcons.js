import React, { useEffect } from 'react';
import { useInView } from 'react-intersection-observer';
import { useAnimation } from 'framer-motion';
import PropTypes from 'prop-types';
import Icon from 'components/atoms/Icon';
import facebook from 'assets/svgs/icon_facebook.svg';
import linkedin from 'assets/svgs/icon_linkedin.svg';
import animations from './SocialMediaIcons.animations';
import S from './SocialMediaIcons.styles';

const SocialMediaIcons = ({ className }) => {
  const animation = useAnimation();

  const [wrapperRef, inView] = useInView({
    triggerOnce: true,
    threshold: 0.5,
  });

  useEffect(() => {
    if (inView) {
      animation.start('animate');
    }
  }, [animation, inView]);

  const wrapperAnimations = animations.getWrapper(animation);

  return (
    <S.Wrapper ref={wrapperRef} {...wrapperAnimations} className={className}>
      <S.A href="https://facebook.com" aria-label="facebook" target="__blank">
        <Icon src={facebook} />
      </S.A>
      <S.A href="https://linkedin.com" aria-label="linkedin" target="__blank">
        <Icon src={linkedin} />
      </S.A>
    </S.Wrapper>
  );
};

SocialMediaIcons.defaultProps = {
  className: '',
};

SocialMediaIcons.propTypes = {
  className: PropTypes.string,
};

export default SocialMediaIcons;
