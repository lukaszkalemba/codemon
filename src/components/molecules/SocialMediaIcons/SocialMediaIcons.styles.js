import styled from 'styled-components';
import { motion } from 'framer-motion';

const Wrapper = styled(motion.div)`
  display: flex;
  flex-direction: column;
  align-items: flex-end;

  ${({ theme }) => theme.media.laptop`
    position: static;
    transform: none;
    padding: 0;
  `}
`;

const A = styled.a`
  opacity: 0.7;
  transition: opacity 250ms ease-in-out, transform 250ms ease-in-out;
  padding: 8px 0;

  :hover {
    opacity: 1;
    transform: scale(1.05);
  }
`;

export default { Wrapper, A };
