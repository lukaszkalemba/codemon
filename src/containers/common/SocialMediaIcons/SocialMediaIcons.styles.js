import styled from 'styled-components';
import { motion } from 'framer-motion';

const Wrapper = styled(motion.div)`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  position: absolute;
  right: 0;
  top: 0;
  padding: 20px;

  ${({ theme }) => theme.media.tablet`
    padding: 35px 55px;
  `}

  ${({ theme }) => theme.media.laptop`
    position: static;
    transform: none;
    padding: 0;
  `}
`;

const A = styled.a`
  opacity: 0.7;
  transition: opacity 250ms ease-in-out, transform 250ms ease-in-out;
  padding: 8px 0;

  :hover {
    opacity: 1;
    transform: scale(1.05);
  }
`;

export default { Wrapper, A };
