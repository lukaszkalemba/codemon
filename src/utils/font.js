export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
