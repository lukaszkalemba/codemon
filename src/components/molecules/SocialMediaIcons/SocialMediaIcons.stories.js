import React from 'react';
import SocialMediaIcons from './SocialMediaIcons';

const variants = {
  animate: {
    opacity: 1,
    y: 0,
    transition: { duration: 0.8, delay: 0 },
  },
  initial: { opacity: 0, y: 20 },
};

export const socialMediaIcons = () => <SocialMediaIcons variants={variants} />;

export default {
  component: SocialMediaIcons,
  title: 'Molecules|SocialMediaIcons',
};
