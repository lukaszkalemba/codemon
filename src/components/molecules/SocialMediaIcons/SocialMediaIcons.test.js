import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import SocialMediaIcons from './SocialMediaIcons';

const renderSocialMediaIcons = props => {
  const utils = render(<SocialMediaIcons {...props} />);

  return { ...utils };
};

describe('<SocialMediaIcons />', () => {
  it('has proper link elements length', () => {
    const { getAllByRole } = renderSocialMediaIcons();

    expect(getAllByRole('link')).toHaveLength(4);
  });

  it('matches snapshot', () => {
    const { container } = renderSocialMediaIcons();

    expect(container.firstChild).toMatchSnapshot();
  });
});
