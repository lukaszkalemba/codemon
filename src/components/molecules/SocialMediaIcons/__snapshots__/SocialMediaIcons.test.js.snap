// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMediaIcons /> matches snapshot 1`] = `
<div
  class="sc-gtssRu kbYlSS"
  style="opacity: 0; transform: translateY(20px) translateZ(0);"
>
  <a
    aria-label="facebook"
    class="sc-dlnjPT caLAjc"
    href="https://facebook.com"
    target="__blank"
  >
    <div
      class="sc-bdnylx levWdn"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 65 16"
        width="65"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 1H61L42.5 14.5"
          stroke="white"
          stroke-width="2"
        />
      </svg>
    </div>
  </a>
  <a
    aria-label="linkedin"
    class="sc-dlnjPT caLAjc"
    href="https://linkedin.com"
    target="__blank"
  >
    <div
      class="sc-bdnylx levWdn"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 65 16"
        width="65"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 1H61L42.5 14.5"
          stroke="white"
          stroke-width="2"
        />
      </svg>
    </div>
  </a>
</div>
`;
